File: frontend/src/MockTestService.js
const BASE_URL = '/api/mocktests';

export async function getAllTests() {
  const res = await fetch(BASE_URL);
  return res.json();
}

export async function getTestById(id) {
  const res = await fetch(`${BASE_URL}/${id}`);
  return res.json();
}

export async function createTest(test) {
  const res = await fetch(BASE_URL, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(test)
  });
  return res.json();
}

export async function updateTest(id, test) {
  const res = await fetch(`${BASE_URL}/${id}`, {
    method: 'PUT',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(test)
  });
  return res.json();
}

export async function deleteTest(id) {
  const res = await fetch(`${BASE_URL}/${id}`, { method: 'DELETE' });
  return res.text(); // your endpoint returns a plain string
}

File: src/main/java/com/MockTest/service/MockTestService.java
package com.MockTest.service;




	import com.MockTest.entity.MockTest;
	import java.util.List;

	public interface MockTestService {

	    MockTest createTest(MockTest mockTest);

	    List<MockTest> getAllTests();

	    MockTest getTestById(Long id);

	    MockTest updateTest(Long id, MockTest mockTest);

	    void deleteTest(Long id);
	}
